### DISCUSSION TITLE
Group table rows to prevent a pagebreak within?

### DISCUSSION THREAD

**Discussion by Schaefer-Tobias** [typst/typst#5337] [Q&A] [answered]
Hi,

let's assume that we want to display a table that spans multiple pages. Within the table we want to specify some consecutive table rows where a pagebreak should not appear within. 

How can I achieve this?

Best regards
Tobias

**Comment by Schaefer-Tobias** [comment]
I just found a solution. The idea is to put all related rows in their own table and wrap this table inside a `block(breakable: false, ...)` element that is itself placed inside an outer (main) table.

```
// page settings
#set page(
    paper: "a4",
    margin: (top: 2cm, bottom: 2cm, x: 2cm, y: 2cm),
)

// define some helper functions
#let rows(i) = {
  let v = "value_" + str(i)
  {
    table(
      columns: (1fr, 1fr),
      inset: 4pt,
      align: (left, left),
      stroke: none,
      table.cell("param_1"), table.cell(v),
      table.cell("param_2"), table.cell(v),
      table.cell("param_3"), table.cell(v),
      table.cell("param_4"), table.cell(v),
      table.cell("param_5"), table.cell(v),
    )
  }
}

#let group(i) = {
  // this is the magic part
  block(
    breakable: false,
    above: 0pt,
    below: 0pt,
    rows(i)
  )
}

// table
#{
    let n = 25
    let data = ()
    
    // create dummy data and append to array
    for i in range(1, n) {
        data.push(group(i))
    }

    show table.cell: set text(
        12pt,
    )

    table(
        columns: (1fr),
        inset: 0pt,
        align: (left),
        stroke: red + 1.0pt,
        //spread data from array into table
        ..data
    )
}
```

**Comment by PgBiel** [comment, marked as answer]
A rowspan with `breakable: false` won't be broken apart by the layout algorithm, so you can simply add a `table.cell(rowspan: N, breakable: false)` across the rows in your group. Since this requires adding a cell, you can do this in a 0pt-wide column in order to not affect the output. In the future, the idea is to have a simpler way to group rows together, e.g. with some row group primitive.

See also https://github.com/typst/typst/issues/4841

**Reply by Schaefer-Tobias** [reply]
Thank you for the reply. Your solution looks a lot easier!
